### PR TITLE
Refactor UI

### PR DESCRIPTION
**Pull request by nwrogers**

### CODE REVIEW

File: src/resultscallercalleepage.cpp [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+    connect(ui->callerCalleeTableView->selectionModel(), &QItemSelectionModel::currentRowChanged,
+            this, [selectCallerCaleeeIndex] (const QModelIndex& current, const QModelIndex&) {
+                if (current.isValid()) {
+                    selectCallerCaleeeIndex(current);
+                }
+            });
+}
+
+ResultsCallerCalleePage::~ResultsCallerCalleePage()
```
Comment (milianw) [inline src/resultscallercalleepage.cpp]: `... = default;`

instead of the `{}`

File: src/resultsflamegraphpage.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    connect(parser, &PerfParser::topDownDataAvailable,
+            this, [this] (const Data::TopDownResults& data) {
+                ui->flameGraph->setTopDownData(data);
+            });
+
+    connect(ui->flameGraph, &FlameGraph::jumpToCallerCallee,
+            this, &ResultsFlameGraphPage::jumpToCallerCallee);
+}
+
+ResultsFlameGraphPage::~ResultsFlameGraphPage()
```
Comment (milianw) [inline src/resultsflamegraphpage.cpp]: default also

File: src/resultspage.cpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    connect(m_resultsBottomUpPage, &ResultsBottomUpPage::jumpToCallerCallee,
+            this, &ResultsPage::onJumpToCallerCallee);
+    connect(m_resultsTopDownPage, &ResultsTopDownPage::jumpToCallerCallee,
+            this, &ResultsPage::onJumpToCallerCallee);
+    connect(m_resultsFlameGraphPage, &ResultsFlameGraphPage::jumpToCallerCallee,
+            this, &ResultsPage::onJumpToCallerCallee);
+}
+
+ResultsPage::~ResultsPage()
```
Comment (milianw) [inline src/resultspage.cpp]: default also

File: src/resultssummarypage.cpp [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+#include "models/summarydata.h"
+#include "models/hashmodel.h"
+#include "models/costdelegate.h"
+#include "models/treemodel.h"
+#include "models/topproxy.h"
+#include "models/callercalleemodel.h"
+
+namespace {
+QString formatTimeString(quint64 nanoseconds)
```
Comment (milianw) [inline src/resultssummarypage.cpp]: move this to the `util.{h,cpp}` files

File: src/resultstopdownpage.cpp [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    ResultsUtil::setupCostDelegate(topDownCostModel, ui->topDownTreeView);
+    connect(ui->topDownTreeView, &QTreeView::customContextMenuRequested, this, &ResultsTopDownPage::onContextMenu);
+
+    connect(parser, &PerfParser::topDownDataAvailable,
+            this, [this, topDownCostModel] (const Data::TopDownResults& data) {
+                topDownCostModel->setData(data);
+            });
+}
+
+ResultsTopDownPage::~ResultsTopDownPage()
```
Comment (milianw) [inline src/resultstopdownpage.cpp]: default

File: src/startpage.cpp [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+    connect(ui->openFileButton, &QPushButton::clicked,
+            this, &StartPage::openFileButtonClicked);
+
+    ui->openFileButton->setFocus();
+
+    updateBackground();
+}
+
+StartPage::~StartPage()
```
Comment (milianw) [inline src/startpage.cpp]: default

**Comment by milianw** [comment]
can you rebase this now, or should I do this for you?

**Comment by nwrogers** [comment]
Refactored the Record GUI and merged all other changes

**Comment by milianw** [comment]
thanks Nate!

**Pull request merged**